evap/results/views.py

```diff
@@ -25,6 +25,8 @@ def index(request):
     courses = [course for course in courses if course.can_user_see_course(request.user)]
 
     for course in courses:
+        if course.is_single_result:
+            course.single_result_rating_result = get_single_result_rating_result(course)
         course.distribution = calculate_average_distribution(course)
         course.avg_grade = distribution_to_grade(course.distribution)
```

Comment:
for single results, i think it should be `course.avg_grade should = course.single_result_rating_result.average` and `course.distribution` shouldn't be written at all.

either we decide to get the results of single results from `calculate_average_distribution` via some hard-coded indexing into the datastructure, or we leave it like you proposed, then i would assert in `calculate_average_distribution` and `collect_results` that they are not used for single results.